### PR TITLE
login web handler

### LINKED ISSUE #155: Login web Handler
User login API implementation.
- Perform login using API, where username and password given as request to login API.
- Return appropriate response to login request.
- Response may include:
  - Login status: Succeed/ Failed
  - Status code: 200/ 401

### PR DESCRIPTION
**Pull request by dipikakoli26**
## Description
Implemented login rest API.
- API accepts user request for login with login credentials - `username` and `password`.
- If the user credential matches a user is authenticated user and success response is sent back.
- If user is not authenticated failure response is sent back.

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Project Configuration
- [ ] Repository Configuration
- [ ] Documentation update - Code guide/ docs (_like, javadoc or api_)
- [ ] Other (please describe):

## Screenshots / GIFs / Media (if applicable)
- NA

## How Has This Been Tested?/ Test Plan
- Send HTTP POST request with `username` and `password` through any Rest API client such as Postman or Swagger.
- Get HTTP response with login status, failure or success. 

## Checklist:
- [x] My code follows the project’s [coding guidelines](./../../docs/guidelines/coding-guides/). 
- [x] I have performed a self-review of my own code.
- [x] I have commented on my code where necessary.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [ ] I have added tests that verify the functionality of the code (if applicable).
- [ ] All new and existing tests pass.

## How to use/ enable
- HTTP request body
  `content-type`:`application/json`
  ```json
  {
    "username": "ajayg2808",
    "password": "ajay1234"
  }
  ```

- HTTP response body
  `content-type`:`application/json`
  Success: `Status: 200 OK`
  ```json
  {
    "id": 1,
    "name": "Ajay Gaikwad",
    "dob": null,
    "address": null,
    "moNo": "9156974513",
    "city": "Kaij",
    "pincode": 431123,
    "username": "ajayg2808",
    "createdAt": "2024-12-16",
    "enabled": false
  }
  ```
  Failed: `Status: 401 Unauthorized`
  ```json
  {
    "Username": "ajayg2808",
    "message": "Invalid username or password"
  }
  ```

## Additional Notes
- NA

## Closing issues
closes #155

### CODE REVIEW

**Comment by ajayg2808** [comment]
Follow **Pull request template**.
Title and description of pull request must be proper, concise and brief overview about what, why and how the task is completed and how to validate it.

**Comment by ajayg2808** [comment]
Do not add labels to pull request until and unless it do not have issue for it.

File: Arc-i-Tech/src/main/java/com/arcitech/service/UserService.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import com.arcitech.repository.UserRepository;
+import com.arcitech.exception.UserAlreadyExistsException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserService {
+
+    @Autowired
+    private UserRepository userRepository;
```
Comment (ajstyle2410) [inline Arc-i-Tech/src/main/java/com/arcitech/service/UserService.java]: this class not present in a package inside

Comment (ajstyle2410) [inline Arc-i-Tech/src/main/java/com/arcitech/service/UserService.java]: this class not present in a package inside

File: Arc-i-Tech/src/main/java/com/arcitech/service/UserService.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Autowired
+    private UserAuthRepository userAuthRepository;
+
+    public User registerUser(User user, UserAuth userAuth) {
+        // Check if username already exists in the UserAuth table
+        if (userAuthRepository.findByUsername(userAuth.getUsername()).isPresent()) {
+            throw new UserAlreadyExistsException("Username " + userAuth.getUsername() + " already exists");
```
Comment (ajstyle2410) [inline Arc-i-Tech/src/main/java/com/arcitech/service/UserService.java]: UserAlreadyExistsException are  not recognize the project inside

File: Arc-i-Tech/src/main/java/com/arcitech/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        // Save User entity and link it with UserAuth
+        User savedUser = userRepository.save(user);
+        userAuth.setUser(savedUser);
+        userAuthRepository.save(userAuth);
```
Comment (ajstyle2410) [inline Arc-i-Tech/src/main/java/com/arcitech/service/UserService.java]: save method not work properly

File: Arc-i-Tech/src/main/java/com/arcitech/controllers/UserController.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    @PostMapping("/register")
+    @ResponseStatus(HttpStatus.CREATED)
+    public UserAuth loginUser(@RequestBody UserAuth userAuth) {
+        return userService.findByUserAndPassword(userAuth);
```
Comment (ajstyle2410) [inline Arc-i-Tech/src/main/java/com/arcitech/controllers/UserController.java]: findByUserAndPassword are not present in a  service inside

**Review by ajstyle2410 — CHANGES_REQUESTED** [review]

File: Arc-i-Tech/src/main/java/com/arcitech/service/UserService.java (hunk 1 above)
Comment (ajstyle2410) [reply Arc-i-Tech/src/main/java/com/arcitech/service/UserService.java]: resolved it

**Comment by ajayg2808** [comment]
@ajstyle2410,

- [x] Resolve merge conflicts.
- [x] Update PR template 

Let me know, once it is done.

**Comment by ajstyle2410** [comment]
> @ajstyle2410,
> 
> * [x]  Resolve merge conflicts.
> * [x]  Update PR template
> 
> Let me know, once it is done.

Conflict resolved

**Comment by ajayg2808** [comment]
> > @ajstyle2410,
> > 
> > * [x]   Resolve merge conflicts.
> > * [x]   Update PR template
> > 
> > Let me know, once it is done.
> 
> Conflict resolved

PR template is not updated yet.
Your PR still is linked to project. I am not going to review it.

**Comment by ajstyle2410** [comment]
@ajayg2808  update the pull request changes the pull request  issue number

**Review by ajayg2808 — APPROVED** [review]

**Pull request merged**